packages/devtools_app/lib/src/performance/performance_model.dart

```diff
@@ -917,43 +917,26 @@ abstract class TimelineEvent extends TreeNode<TimelineEvent>
 class SyncTimelineEvent extends TimelineEvent {
   SyncTimelineEvent(TraceEventWrapper firstTraceEvent) : super(firstTraceEvent);
 
-  bool get isUiEventFlow => uiFrameNumber != null;
-
-  bool get isRasterEventFlow => rasterFrameNumber != null;
-
-  int get uiFrameNumber {
-    if (_uiFrameNumber != null) return _uiFrameNumber;
-    final animatorBeginFrameEvent = firstChildWithCondition(
-        (TimelineEvent event) => event.name.contains(uiEventName));
-    if (animatorBeginFrameEvent != null) {
-      try {
-        return int.parse(animatorBeginFrameEvent
-            .traceEvents.first.event.args[TraceEvent.frameNumberArg]);
-      } catch (_) {
-        return null;
-      }
-    }
-    return null;
+  int get frameNumberFromArgs {
+    final frameNumber = traceEvents.first.event.args[TraceEvent.frameNumberArg];
```

Comment:
Does the getter need to clarify where it is from?
Can this just be called
frameNumber.
You might want to still cache the value as int.tryParse is somewhat expensive.

Reply:
cached and renamed